apps/dashboard/app/helpers/batch_connect/sessions_helper.rb

```diff
@@ -214,4 +214,14 @@ def connection_tabs(id, tabs)
       end
     end
   end
+
+  def adjusted_poll_delay(sessions)
+    return if sessions.all? { |s| s.completed? }
+
+    if sessions.any? { |s| s.queued? || s.starting? }
+      ConfigurationSingleton::MIN_POLL_DELAY
+    else
+      Configuration.bc_sessions_poll_delay
+    end
```

Comment:
OK - last thing (I hope :pray) - This actually speeds up polling which I don't think we want.  Maybe I suggested this previously? I don't know, but in any case, I'm now questioning it.

Seems like we should just poll at `bc_sessions_poll_delay` or not at all.  This whole endeavor is to poll _less_, so let's just do that. I don't think we need to poll more under certain conditions, instead just stop polling when they're all completed.

Reply:
Adding on to the idea of polling less or stopping the polling, I think it would be quite nice if we would check whether the tab/window is visible at all ([document.visibilityState](https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilityState)) to potentially skip or delay the polling further.
I would imagine that the typical scenario in OOD is that the user click the connect button and are sent to the app session in a new tab. Then the tab with the interactive app sessions would be hidden, which we can detect. Depending on the rest of the logic, we may or may not want to poll again once the tab/window is visible again (`visibilitychange` event).